django/db/models/manager.py

```diff
@@ -81,11 +82,10 @@ def check(self, **kwargs):
     @classmethod
     def _get_queryset_methods(cls, queryset_class):
         def create_method(name, method):
+            @wraps(method)
```

Comment:
A regression test is required.

Reply:
@felixxm What exactly should the test case test?
Whether the decorated function has the same name/doc/signature as the original function? If yes, then adding such a test case is equivalent to testing whether `wraps` is working as expected or not, which is obviously not required since it's a built-in Python function.

Actually, in the current state, i.e. in the absence of `wraps`, it would have made sense to add test cases ensuring the decorated function has the right name/doc. With `wraps` it seems pointless.

Please correct me if I am missing anything here. Thanks.

Reply:
I would test exactly what you reported, e.g.
```python
        self.assertEqual(
            str(inspect.signature(Article.objects.bulk_create)),
            "(objs, batch_size=None, ignore_conflicts=False, update_conflicts=False, "
            "update_fields=None, unique_fields=None)",
        )
```

Reply:
@felixxm If we hard code the signature(or any other attribute), then this test case need to be updated every time the original function's signature, docstring, etc are updated. To avoid that can I do the below?
```python3
    def test_manager_methods_attributes(self):
        for method in self.QUERYSET_PROXY_METHODS:
            manager_method = getattr(Article.objects, method)
            queryset_method = getattr(models.QuerySet, method)

            self.assertEqual(
                str(inspect.signature(manager_method)),
                str(inspect.signature(queryset_method)).replace('self, ', '').replace('self', '')
            )
            
            self.assertEqual(
                manager_method.__doc__,
                queryset_method.__doc__
            )

            self.assertEqual(
                manager_method.__annotations__,
                queryset_method.__annotations__
            )
```

Reply:
> ... then this test case need to be updated every time the original function's signature, docstring, etc are updated.

That's not true. Proposed assertion compares only a signature:

> ```python
>         self.assertEqual(
>             str(inspect.signature(Article.objects.bulk_create)),
>             "(objs, batch_size=None, ignore_conflicts=False, update_conflicts=False, "
>             "update_fields=None, unique_fields=None)",
>         )
> ```

and I don't see a big issue in updating it. We don't change signatures so often.

> Moreover, our goal is to ensure manager methods show the same signature and docstrings as the queryset methods and not a specific string.

It's exactly the same in this case. I'd prefer to compare with the exact value not the result of another function.